### PR TITLE
[Android] support SNAP in Android

### PR DESCRIPTION
**Pull request by jaeyun-jung**
In Java APIs, handle multiple model files and custom option to support SNAP.
Define enum for supported NNFW and implement native functions.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
PR to run SNAP in Java APIs

waiting for other PRs, need test in target device

File: api/capi/src/nnstreamer-capi-single.c [hunk 1]
```diff
@@ -374,129 +374,173 @@ ml_single_set_info_in_handle (ml_single_h single, gboolean is_input,
...
+  ml_tensors_info_initialize (&single_h->in_info);
+  ml_tensors_info_initialize (&single_h->out_info);
+  g_mutex_init (&single_h->mutex);
+  g_cond_init (&single_h->cond);
+
+  single_h->thread = g_thread_try_new (NULL, invoke_thread, (gpointer) single_h, &error);
+  if (single_h->thread == NULL) {
+    ml_loge ("Failed to create the invoke thread, error: %s.", error->message);
+    g_clear_error (&error);
+    ml_single_close (single_h);
```
Comment (dongju-chae) [inline api/capi/src/nnstreamer-capi-single.c]: Q. `g_thread_join (NULL)` (called in `ml_single_close()`) is safe?

Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c]: g_thread_join() internally checks null ptr.
```
g_thread_join (GThread *thread)
{
  ...
  g_return_val_if_fail (thread, NULL);
```

**Comment by jaeyun-jung** [comment]
I will mark the negative testcases and check null thread ptr in single handle.

**Pull request merged**